### ISSUE TITLE
chatglm2 sft微调报错ValueError

### ISSUE THREAD

**Issue by zoepo** [shibing624/MedicalGPT#118]
### Describe the bug
使用chatglm2的模型，`template_name`尝试过 chatglm、vicuna、alpaca及chatglm2，数据集用的`alpaca_gpt4_data_zh`，在`train_dataset.shuffle().map`的部分报错，报错如下
![image](https://github.com/shibing624/MedicalGPT/assets/121851857/c44e2a99-f242-4892-998d-39eb5c01d0cf)

**Comment by zoepo** [comment]
> alpaca_gpt4_data_zh是alpaca格式，需要通过https://github.com/shibing624/MedicalGPT/blob/main/convert_alpaca.py 转换为ShareGPT格式

转换后通过了，但是出现新的问题，`transformers`版本是`4.30.2`
![image](https://github.com/shibing624/MedicalGPT/assets/121851857/9e0d7d65-9d44-494c-acac-682e205c37df)

**Comment by shibing624** [comment]
这个是chatglm2官方代码的bug，我提交了修改方法：https://huggingface.co/THUDM/chatglm2-6b/discussions/67

**Comment by valkryhx** [comment]
> alpaca_gpt4_data_zh是alpaca格式，需要通过https://github.com/shibing624/MedicalGPT/blob/main/convert_alpaca.py 转换为ShareGPT格式

这样是把指令微调转化成单轮对话 从而兼容训练代码了吗？

**Comment by shibing624** [comment]
嗯
